bascule/key/mocks.go

```diff
@@ -0,0 +1,80 @@
+package key
+
+import (
+	"github.com/stretchr/testify/mock"
+)
+
+// MockResolver is a stretchr mock for Resolver.  It's exposed for other package tests.
+type MockResolver struct {
```

Comment:
The MockResolver and MockCache types needs to be updated with the new Resolver and Cache method signatures.  In order to ensure these types implement those interfaces, you can do this:

```var _ Resolver = (*MockResolver)(nil)```

That will break at compiletime (in the current version of this code especially) since the signature of `ResolveKey` changed.  It's a nice way of ensuring that types implement interfaces when those types aren't directly used in the production code of a package.